x-pack/plugins/security_solution/public/common/hooks/eql/api.ts

```diff
@@ -29,3 +40,64 @@ export const validateEql = async ({
     signal,
   });
 };
+
+interface AggsParams extends EqlValidationRequest {
+  data: DataPublicPluginStart;
+  interval: Unit;
+  fromTime: string;
+  toTime: string;
+  signal: AbortSignal;
+}
+
+export const getEqlPreview = async ({
+  data,
+  index,
+  interval,
+  query,
+  fromTime,
+  toTime,
+  signal,
+}: AggsParams): Promise<EqlPreviewResponse> => {
+  try {
+    const response = await data.search
+      .search<EqlSearchStrategyRequest, EqlSearchStrategyResponse<EqlSearchResponse<Source>>>(
+        {
+          params: {
+            // @ts-expect-error allow_no_indices is missing on EqlSearch
+            allow_no_indices: true,
+            index: index.join(),
+            body: {
+              filter: {
+                range: {
+                  '@timestamp': {
```

Comment:
should this be using the timestamp override from the rule if available?

Reply:
At first I thought, yes, but we got asked to move the timestamp override field to step 2 and not take it into account. But maybe we want to reconsider? @marrasherrier @dontcallmesherryli